testing/src/main/java/org/oppia/android/testing/TextInputAction.kt

```diff
@@ -0,0 +1,38 @@
+package org.oppia.android.testing
+
+import android.content.Context
+import androidx.annotation.StringRes
+import com.google.android.material.textfield.TextInputLayout
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.TypeSafeMatcher
+import javax.inject.Inject
+
+class TextInputAction @Inject constructor(
+  val context: Context
+) {
+  fun hasErrorText(@StringRes expectedErrorTextId: Int): Matcher<TextInputLayout> {
+    return object : TypeSafeMatcher<TextInputLayout>() {
+      override fun matchesSafely(textInputLayout: TextInputLayout): Boolean {
+        val expectedErrorText = context.resources.getString(expectedErrorTextId)
+        return (textInputLayout).error == expectedErrorText
+      }
+
+      override fun describeTo(description: Description) {
+        description.appendText("TextInputLayout's error")
+      }
+    }
+  }
+
+  fun hasNoErrorText(): Matcher<TextInputLayout> {
```

Comment:
Add kdoc

Reply:
Make these functions static to access these methods directly. 

Reason when we can/cannot use companion object in such case - 
We use companion object/make function static when our helper class has no dependency. For example, the current `context` is the dependency, but we don't need it anymore so we can make these functions static. \

Reply:
Done